server/src/main/java/tabletop/services/SwissTournamentService.java

```diff
@@ -0,0 +1,143 @@
+package tabletop.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import tabletop.controllers.TournamentController;
+import tabletop.domain.match.tournament.Pair;
+import tabletop.domain.match.tournament.Tournament;
+import tabletop.domain.match.tournament.TournamentFinalResult;
+import tabletop.domain.match.tournament.swiss.SwissPlayerResult;
+import tabletop.domain.match.tournament.swiss.SwissTournamentProcess;
+import tabletop.domain.user.User;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+@Service
+public class SwissTournamentService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SwissTournamentService.class);
+
+
+    public List<Pair<User>> getInitialPairs(SwissTournamentProcess swissTournamentProcess) {
+        if (!swissTournamentProcess.isRanked()) {
+            // random seeding
+            Collections.shuffle(swissTournamentProcess.getUsers(), new Random(System.nanoTime()));
+        }
+
+        // initial pairing
+        List<Pair<User>> pairs = new LinkedList<>();
+        for (int i = 0; i < swissTournamentProcess.getUsers().size(); i += 2) {
+            if (i + 1 < swissTournamentProcess.getUsers().size()) {
+                User user1 = swissTournamentProcess.getUsers().get(i);
+                User user2 = swissTournamentProcess.getUsers().get(i + 1);
+                Pair<User> pair = new Pair<>(user1, user2);
+                pairs.add(pair);
+                swissTournamentProcess.getResultByUser(user1).ifPresent(result -> setInitialResult(user2, result));
```

Comment:
can this return an empty optional in here? if not, you could just use .get() instead of ifPresent